base/shell/explorer/explorer.cpp

```diff
@@ -87,6 +87,45 @@ HideMinimizedWindows(IN BOOL bHide)
 }
 #endif
 
+static BOOL
+IsExplorerSystemShell()
+{
+    HKEY hKeyWinlogon;
+	WCHAR szShell[256];
+	WCHAR szPath[MAX_PATH];
+    LPWSTR szExplorer = NULL;
+	DWORD dwType;
+	DWORD dwBufferSize = sizeof(szShell);
+    BOOL bIsSystemShell = TRUE;
+
+    if (!GetModuleFileName(NULL, szPath, MAX_PATH))
+        return FALSE;
+
+    szExplorer = PathFindFileName((LPWSTR)szPath);
+
+	if (RegOpenKeyEx(HKEY_LOCAL_MACHINE, L"Software\\Microsoft\\Windows NT\\CurrentVersion\\Winlogon",
+		0, KEY_READ, &hKeyWinlogon) != ERROR_SUCCESS)
+    {
+        // No registry access.
+        bIsSystemShell = TRUE;
+    }
+    else
+    {
+        if (RegQueryValueEx(hKeyWinlogon, L"Shell", 0, &dwType,
+            (LPBYTE)szShell, &dwBufferSize) == ERROR_SUCCESS)
+        {
+            if (StrStrI(szShell, szExplorer))
+                bIsSystemShell = TRUE;
+            else
+                bIsSystemShell = FALSE;
+        }
+
+        RegCloseKey(hKeyWinlogon);
+    }
+
+	return bIsSystemShell;
```

Comment:
Fixing indentation (replacing tab by 4 spaces):
```suggestion
    return bIsSystemShell;
```